gitlab.go

```diff
@@ -520,11 +520,11 @@ func (c *Client) setBaseURL(urlStr string) error {
 	return nil
 }
 
-// NewRequest creates an API request. A relative URL path can be provided in
-// path, in which case it is resolved relative to the base URL of the Client.
-// Relative URL paths should always be specified without a preceding slash. If
-// specified, the value pointed to by body is JSON encoded and included as the
-// request body.
+// NewRequest creates an API request. An optional relative URL path can be
```

Comment:
Is URL path really optional? What happens if you provide an empty string?

Reply:
Then it constructs a request that targets the root of the API (see base URL). Not sure if that is a useful endpoint that returns anything useful, but it's perfectly valid to construct a request like that.